docs/spec/openapi/public-docs.yaml

```diff
@@ -0,0 +1,5703 @@
+        must be followed to configure the correct permissions for APM access.
+      operationId: getRumAgentConfig
+      tags:
+        - agent config
+      responses:
+        '200':
+          description: Successful response.
+  /intake/v2/events:
+    post:
+      summary: Send APM agent events
+      description: >
+        NOTE: Most users do not need to interact directly with the events intake
+        API. Agents communicate with the APM Server by sending events--captured
+        pieces of information--in an HTTP request. Events can be transactions,
+        spans, errors, or metrics. Each event is sent as its own line in the
+        HTTP request body, which is known as newline delimited JSON (NDJSON).
+        With NDJSON, agents can open an HTTP POST request and use chunked
+        encoding to stream events to the APM Server as soon as they are recorded
+        in the agent. This makes it simple for agents to serialize each event to
+        a stream of newline delimited JSON. The APM Server also treats the HTTP
+        body as a compressed stream and thus reads and handles each event
+        independently. See the [APM data
+        model](https://www.elastic.co/guide/en/observability/current/apm-data-model.html)
+        to learn more about the different types of events.
+      operationId: postEventIntake
+      tags:
+        - event intake
+      requestBody:
+        content:
+          application/ndjson:
+            schema:
+              anyOf:
+                - $ref: '#/components/schemas/ErrorEvent'
+                - $ref: '#/components/schemas/MetadataEvent'
+                - $ref: '#/components/schemas/MetricSetEvent'
+                - $ref: '#/components/schemas/SpanEvent'
+                - $ref: '#/components/schemas/TransactionEvent'
+            examples:
+              postMetadataEventIntakeExample:
+                $ref: '#/components/examples/metadataEventIntakeRequestExample'
+              postMetricSetEventIntakeExample:
+                $ref: '#/components/examples/metricSetEventIntakeRequestExample'
+      responses:
+        '202':
+          description: Successful response; all events succeeded.
+  /intake/v2/rum/events:
+    post:
+      summary: Send RUM events
+      description: >
+        NOTE: Most users do not need to interact directly with the events intake
+        API.
+      operationId: postRumEventIntakeV2
+      tags:
+        - event intake
+      requestBody:
+        content:
+          application/ndjson:
+            schema:
+              anyOf:
+                - $ref: '#/components/schemas/ErrorEvent'
+                - $ref: '#/components/schemas/MetadataEvent'
+                - $ref: '#/components/schemas/MetricSetEvent'
+                - $ref: '#/components/schemas/SpanEvent'
+                - $ref: '#/components/schemas/TransactionEvent'
+      responses:
+        '202':
+          description: Successful response; all events succeeded.
+  /intake/v3/rum/events:
```

Comment:
The request body for v3 rum intake is actually different, but I don't think we had documented this correctly in the past. The spec is available in https://github.com/elastic/apm-server/tree/main/docs/spec/rumv3. 
I am not convinced it is worth documenting this API though, afaik it is barely used.